### ISSUE TITLE
Changing PWD in the file viewer doesn't change PWD in your python session

### ISSUE THREAD

**Issue by hernamesbarbara** [yhat/rodeo#331]
**Steps**

* double click folders in the file viewer buffer (bottom right)
* navigate to a folder on your computer that has a python script and some other txt or csv file
* try to read the txt or csv file from your python session by either running a LOC in your script or by running a read file command from the python shell buffer

**expected**

* since you told Rodeo to change directories to this new folder, the user expects to be able to read files relative to that new working directory

**observed**

* txt or csv file is not found by rodeo since the python session has not changed to the working directory you selected in the file viewer

![image](https://cloud.githubusercontent.com/assets/738191/15933374/89513b60-2e2d-11e6-892d-2eff9a192699.png)

![image](https://cloud.githubusercontent.com/assets/738191/15933409/ac39919a-2e2d-11e6-9435-72d608d5e360.png)

![image](https://cloud.githubusercontent.com/assets/738191/15933398/a9496744-2e2d-11e6-881e-a9efdc869258.png)

**Comment by TakenPilot** [comment]
Is it an expected behavior for the file viewer and the console to be linked?

**Comment by hernamesbarbara** [comment]
we were talking about it in slack and it seems this is a new feature concept rather than a bug. the behavior i described above in the issue is currently working as designed.

the new feature would give the user more control over how this behaves. something like this:

![image](https://cloud.githubusercontent.com/assets/738191/15936318/943d8b86-2e38-11e6-9a6c-89ab087a983d.png)

**Comment by ben519** [comment]
I get & respect the idea, but, for what it's worth here's my case against it.

I think most people including myself associate projects with some sort of top project directory.  For example, when I start a new [Kaggle](https://www.kaggle.com/) competition, I create a "CompeitionX" directory with a bunch of subdirectories, python scripts, data files, etc. but I always work under the assumption that my working directory is /Users/Ben/Documents/Kaggle/CompetitionX.  So, at any point I know I can run `df = pd.read_csv("Data/somedata.csv")` and that will work.  Meanwhile, I use Rodeo's file viewer to explore my files and mostly to dig up python scripts.  If that navigation changed my current working directory, it would constantly and annoyingly break all the code in my scripts like `df = pd.read_csv("Data/somedata.csv")`.

But there will an option to turn this behavior off, so what's the big deal?

1) I think that using this feature encourages bad programming practices.
2) Anyone who doesn't realize the feature exists and is turned on will experience some big headaches
3) [By far the most important].  I don't want to see Rodeo become Spyder - an overly complex IDE with 1000 features that nobody wants.  I support feature enhancements but I think y'all should put some serious weight on the complexity cost of adding a new feature vs. the benefit of adding it.  I don't think this feature's benefit is worth the complexity it adds.
4) There are other features I think are much more deserving of attention. E.g. auto-closing parentheses and brackets, autocompletion within the Python Shell, etc.

With that said, what would be nice is if Rodeo simply displayed the current working directory, similar to RStudio.

<img width="616" alt="screen shot 2016-06-09 at 5 28 27 pm" src="https://cloud.githubusercontent.com/assets/723099/15948665/a68c58ce-2e67-11e6-914c-36717ef5168b.png">

**Comment by TakenPilot** [comment]
I agree with everything Ben said.  So much so that I'm going to repeat his ideas and add my own take as well.  😄 

If we did do a feature like this, I think we should find a design that satisfies the following:

1.  Should handle the idea of multiple terminals, or rather that the terminal might have been dragged into another window, is behind a different tab, or not immediately visible beside the file viewer.

2.  Should handle the idea that the project might have a ​_directory structure_​, so that different files are in different directories.  For example, someone might store their python files in a different location than their data.  If so, they navigate to their python files, double-click on those to open them, then navigate to the data files to open those.  That our file viewer only shows a single level deep is ​_terrible_​ and needs to change.

3.  Magic is an anti-pattern.  Having the user do an action in one part of the UI while something changes in a completely different part of the interface might as well be invisible, so it falls under the category of magic.  Better would be for them to do an action to opt into whatever you're trying to do, like a button to change the working directory of the terminal to the current root of the file viewer or to a particular folder that is selected.

As for adding an extra setting for this stuff -- not everything should be a setting, and adding extra settings for every semi-feature becomes a form of ​_design by committee_​.  Similar to the reason why we shouldn't be trying to be Sublime, or try to pull in every plugin that Sublime and Atom supports:  we just have better things to do that will contribute to Rodeo becoming a widely used tool for data science.

**Comment by glamp** [comment]
@hernamesbarbara @ben519 Take a look at the 2.4.0 release. It answers a lot of the questions outlined above, but in a different way. Personally, I found it much easier to use and understand! Check it out 👍 

![image](https://cloud.githubusercontent.com/assets/1409333/18954940/37da6bdc-8623-11e6-83fa-6174b06fdd6f.png)
![image](https://cloud.githubusercontent.com/assets/1409333/18954931/2f0d6b8a-8623-11e6-92ae-874ddbc9e9e8.png)
